caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.form.js

```diff
@@ -977,6 +979,10 @@
       this._toggleMAQToggle(true)
     }
 
+    this.setTapSyncEndpoint = function() {
+      this.configuration.options.tapSyncEndpoint = this.getDataTrain()._registryClient.getLastEndpoint()
```

Comment:
This looks brittle.  The application is _borrowing_ the registry client from the Data Train.  Is this safe?  What if the DataTrain fails?  Also, isn't the `_registryClient` _actually_ the TAP client?

Reply:
Reworked _registryClient so it's now called _searchTapClient - including changing the namespace on SearchTapClient so it reflects it's job better. There's now one instance of SearchTapClient, instantiated in cadc.search.app.js. A reference to it is passed to the form and hierarchy instances. 

Form instances refer to the SearchTapClient instance for the last end point still, but it has it's own reference to it now. As long as the initial TAP schema call at page load succeeds, there will be a value set.  (If that doesn't succeed, the code isn't going to get that far anyway...)